src/gen/validation_error.rs

```diff
@@ -0,0 +1,99 @@
+use crate::allocator::{Allocator, NodePtr, SExp};
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum ErrorCode {
+    NegativeAmount,
+    InvalidConditionOpcode,
+    InvalidParentId,
+    InvalidPuzzleHash,
+    InvalidPubkey,
+    InvalidMessage,
+    InvalidCondition,
+    InvalidCoinAmount,
+    InvalidCoinAnnouncement,
+    InvalidPuzzleAnnouncement,
+    AssertHeightAbsolute,
+    AssertHeightRelative,
+    AssertSecondsAbsolute,
+    AssertSecondsRelative,
+    AssertMyAmountFailed,
+    AssertMyPuzzlehashFailed,
+    AssertMyParentIdFailed,
+    AssertMyCoinIdFailed,
+    AssertPuzzleAnnouncementFailed,
+    AssertCoinAnnouncementFailed,
+    ReserveFeeConditionFailed,
+    DuplicateOutput,
+    DoubleSpend,
+    CostExceeded,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct ValidationErr(pub NodePtr, pub ErrorCode);
+
+pub fn checked_add(lhs: u64, rhs: u64, n: NodePtr, e: ErrorCode) -> Result<u64, ValidationErr> {
```

Comment:
Something similar at https://doc.rust-lang.org/std/primitive.u64.html#method.checked_add

Reply:
You could use that in conjunction with `.ok_or` https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

Reply:
done